### ISSUE TITLE
Question: will setAppElement(document.body) work as expected?

### ISSUE THREAD

**Issue by vincerubinetti** [reactjs/react-modal#718]
As I understand it, `setAppElement()` is needed so everything in the background, behind the modal, will be set to "hidden" for screen readers. The documentation specifies that `setAppElement()` should be applied to a React app (top-level) element.

However, what if we are developing a smaller React app that gets injected to a larger existing (third-party) page that does not use React. How do we make sure the rest of the whole page gets "hidden" when our modal is open? I would think I would just do `ReactModal.setAppElement(document.body)`, but I'm not sure if the plugin was designed to handle this.

I cannot test this because I do not own a Screen Reader. Also could not find anything about this in the docs.

**Comment by diasbruno** [comment]
Hi @vincerubinetti, `ReactModal.setAppElement(document.body)` is not a good idea because the modal will be a children of the element that has the `aria-hidden`.

**Comment by jrdevforlife** [comment]
So what is the right way to do it then?

**Comment by paul-mesnilgrente** [comment]
I am guessing that we should have a div inside the body like this:

```html
<body>
  <div id="root">
     Put all your html here
  </div>
</body>
```

and give it root?

**Comment by diasbruno** [comment]
The idea is that the `div` where all modals will be "mounted" on is different from the rest of the application.

```html
<body>
  <div id="app" />
  <div id="modals" />
</body>
```

So react-modal can do

```html
<body>
  <div id="app" aria-hidden="true" />
  <div id="modals" />
</body>
```
